packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -18,17 +25,53 @@ const CONTEXT_MENU_NAME = 'ContextMenu';
 type ContextMenuContextValue = {
   open: boolean;
   onOpenChange(open: boolean): void;
+  onRootContentRef(node: HTMLDivElement): void;
 };
 
 const [ContextMenuProvider, useContextMenuContext] = createContext<ContextMenuContextValue>(
   CONTEXT_MENU_NAME
 );
 
-const ContextMenu: React.FC<{ onOpenChange?(open: boolean): void }> = (props) => {
+type ContextMenuCommonOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  onOpenChange?(open: boolean): void;
+};
```

Comment:
As we discussed – these props won't accurately reflect the interface as it varies during runtime depending on the nested level of the menu.